samples/modules/lvgl/touch_draw/src/main.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2024 Martin Kiepfer <mrmarteng@teleschirm.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <lvgl.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+#include <lvgl_input_device.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(app);
+
+#ifdef CONFIG_GPIO
+struct button_callback_context {
+	struct gpio_callback callback;
+	lv_obj_t *canvas;
+};
+static struct gpio_dt_spec button_gpio = GPIO_DT_SPEC_GET_OR(DT_ALIAS(sw0), gpios, {0});
+
+static void button_isr_callback(const struct device *port, struct gpio_callback *cb, uint32_t pins)
+{
+	ARG_UNUSED(port);
+	ARG_UNUSED(pins);
+
+	struct button_callback_context *ctx =
+		CONTAINER_OF(cb, struct button_callback_context, callback);
+
+	lv_canvas_fill_bg(ctx->canvas, lv_color_black(), LV_OPA_TRANSP);
+}
+#endif /* CONFIG_GPIO */
+
+static void lv_draw_pressing_callback(lv_event_t *e)
+{
+	lv_indev_t *indev = lv_indev_get_act();
```

Comment:
You probably want to use the method provided by this header instead [https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/include/lvgl_input_device.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/lvgl/include/lvgl_input_device.h). Reason is that this method returns the current indev being processed. There is no guarantee that this is the pointer device if there are several of them.